src/models/dashboard.model.ts

```diff
@@ -3,13 +3,31 @@ import mongoose, { Schema, Document } from 'mongoose';
 
 /** Mongoose button interface declaration */
 export interface Button {
-  text: string;
-  href: string;
-  hasRoleRestriction: boolean;
-  roles: string[];
-  variant: string;
-  category: string;
-  openInNewTab: boolean;
+  general: {
```

Comment:
@TaiKamilla 

please keep it as it was before

the "action" / "general" is just an UI thing, I don't want to add a nested level to store elements like text, href, etc

but for the action it is fine, you can keep the nesting